### PR TITLE
rename deleteCollectionBySlug to unloadCollectionBySlug

### LINKED ISSUE #5238: In reducers/collections.js, rename deleteCollectionBySlug to unloadCollectionBySlug
As discussed at https://github.com/mozilla/addons-frontend/pull/5186#discussion_r193733916, the current name of the method is a bit misleading, and it should be changed.

### PR DESCRIPTION
**Pull request by championshuttler**
Fixes #5238

### CODE REVIEW

File: tests/unit/amo/sagas/test_collections.js [hunk 1]
```diff
@@ -537,7 +537,7 @@ describe(__filename, () => {
         // Make sure the the collection is not deleted.
```
Comment (bobsilverberg) [inline tests/unit/amo/sagas/test_collections.js]: This comment should be updated to read: `Make sure the the collection is not unloaded.`

File: src/amo/reducers/collections.js [hunk 2]
```diff
@@ -573,7 +573,7 @@ type DeleteCollectionBySlugAction = {|
   payload: {| slug: string |},
 |};
 
-export const deleteCollectionBySlug = (
+export const unloadCollectionBySlug = (
   slug: string
 ): DeleteCollectionBySlugAction => {
```
Comment (bobsilverberg) [inline src/amo/reducers/collections.js]: We also need to rename `DeleteCollectionBySlugAction` to `UnloadCollectionBySlugAction`, and the const `DELETE_COLLECTION_BY_SLUG` should be renamed to `UNLOAD_COLLECTION_BY_SLUG`.

**Comment by championshuttler** [comment]
@bobsilverberg Done  ;)

**Comment by willdurand** [comment]
😍

**Pull request merged**